tests/Integration/Common/IntegrationTestCase.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace DDTrace\Tests\Integration\Common;
+
+use DDTrace\Span;
+use DDTrace\Tracer;
+use DDTrace\Tests\DebugTransport;
+
+use PHPUnit\Framework\TestCase;
+use OpenTracing\GlobalTracer;
+
+
+/**
+ * A basic class to be extended when testing integrations.
+ */
+abstract class IntegrationTestCase extends TestCase
+{
+    /**
+     * @param $fn
+     * @return Span[][]
+     */
+    public function withTracer($fn)
+    {
+        $transport = new DebugTransport();
+        $tracer = new Tracer($transport);
+        GlobalTracer::set($tracer);
+
+        $fn($tracer);
+        $tracer->flush();
+
+        return $transport->getTraces();
+    }
+
+    /**
+     * @param $name string
+     * @param $fn
+     * @return Span[][]
+     */
+    public function inTestScope($name, $fn)
+    {
+        return $this->withTracer(function ($tracer) use ($fn, $name) {
+            $scope = $tracer->startActiveSpan($name);
+            $fn($tracer);
+            $scope->close();
+        });
+    }
+
+    /**
+     * Checks the exact match of a set of SpanAssertion with the provided Spans.
+     *
+     * @param $traces
+     * @param SpanAssertion[] $expectedSpans
+     */
+    public function assertSpans($traces, SpanAssertion ...$expectedSpans)
+    {
+        // First we assert that ALL the expected spans are in the actual traces and no unexpected span exists.
+        $expectedSpansReferences = array_map(function(SpanAssertion $assertion) {
+            return $assertion->getOperationName();
+        }, $expectedSpans);
+        sort($expectedSpansReferences);
+        $tracesReferences = array_map(function(Span $span) {
+            return $span->getOperationName();
+        }, $this->flattenTraces($traces));
+        sort($tracesReferences);
+
+        $this->assertEquals($expectedSpansReferences, $tracesReferences, 'Missing or additional spans.');
+
+        // Then we assert content on each individual received span
+        foreach ($expectedSpans as $ex) {
+            $this->assertSpan($traces, $ex);
+        }
+    }
+
+    /**
+     * Checks that a span expectation is matched in a collection on Spans.
+     *
+     * @param $traces
+     * @param SpanAssertion $exp
+     */
+    public function assertSpan($traces, SpanAssertion $exp)
```

Comment:
Have you considered extracting this method to a class that specifically performs the assertions ?

Reply:
This looks like a great idea, let me look into it.